Description: Adds a build install path
 Brahms uses a variable SYSTEMML_INSTALL_PATH within the code to find
 libraries which it dynamically links at runtime. In the upstream
 code, this path is used as the install path. In the Debian package
 build process, it's important to install in a separate location so that
 files can then be copied into the location defined by SYSTEMML_INSTALL_PATH.
 This patch adds SYSTEMML_BUILD_INSTALL_PATH which does just that.
Forwarded: No need to forward, packaging fix only.
Author: Seb James <seb.james@sheffield.ac.uk>
Last-Update: 2014-02-21
--- a/makefile.brahms
+++ b/makefile.brahms
@@ -137,7 +137,17 @@
 ifeq ($(SYSTEMML_INSTALL_PATH),)
 ${error set SYSTEMML_INSTALL_PATH}
 endif
-PATH_SML=${subst $(SPACE),\$(SPACE),${subst \,/,$(SYSTEMML_INSTALL_PATH)}}
+
+# The "build install path" is a location into which the build will be
+# installed, prior to being installed on the system at
+# SYSTEMML_INSTALL_PATH. This is necessary for the Debian packaing
+# system, so that the debuild process can build, then install.  If
+# this variable is unset, set it to SYSTEMML_INSTALL_PATH.
+ifeq ($(SYSTEMML_BUILD_INSTALL_PATH),)
+SYSTEMML_BUILD_INSTALL_PATH=$(SYSTEMML_INSTALL_PATH
+endif
+
+PATH_SML=${subst $(SPACE),\$(SPACE),${subst \,/,$(SYSTEMML_BUILD_INSTALL_PATH)}}
 PATH_BRM=$(PATH_SML)/BRAHMS
 
 # temp path
